torchok/models/backbones/base_backbone.py

```diff
@@ -1,37 +1,32 @@
-from abc import ABC, abstractmethod
-from typing import List, Tuple, Union, Optional
+from abc import ABC
+from typing import Tuple
+
+from timm.models.features import FeatureHooks
 
 from torchok.models.base import BaseModel
 
 
 class BaseBackbone(BaseModel, ABC):
     """Base model for TorchOk Backbones"""
-    def __init__(self,
-                 in_channels: Optional[Union[int, List[int], Tuple[int, ...]]] = None,
-                 out_channels: Optional[Union[int, List[int], Tuple[int, ...]]] = None,
-                 out_feature_channels: Optional[Union[List[int], Tuple[int, ...]]] = None):
-        """Init BaseBackbone.
 
-        Args:
-            in_channels: Number of input channels.
-            out_features: Number of output channels - channels after forward method.
-            out_feature_channels: Number of output feature channels - channels after forward_features method.
-        """
-        super().__init__(in_channels, out_channels)
-        self._out_feature_channels = out_feature_channels
+    def create_hooks(self):
+        self.stage_names = [i['module'] for i in self.feature_info]
+        self._out_encoder_channels = [i['num_chs'] for i in self.feature_info]
+        hooks = [dict(module=name, type='forward') for name in self.stage_names]
+        self.feature_hooks = FeatureHooks(hooks, self.named_modules())
 
-    @abstractmethod
-    def forward_features(self, *args, **kwargs):
+    def forward_features(self, x):
         """Forward method for getting backbone feature maps.
            They are mainly used for segmentation and detection tasks.
         """
-        pass
+        last_features = self(x)
```

Comment:
This row doesn't need

Reply:
Without it feature hooks will not work. Feature hooks require forward through model to hook up output